src/DetailsView/components/export-dialog-with-local-state.tsx

```diff
@@ -49,11 +49,10 @@ export class ExportDialogWithLocalState extends React.Component<
     };
 
     private generateHtml = () => {
-        const exportData = this.props.htmlGenerator(this.state.exportDescription);
-        this.setState({
-            exportData,
+        this.setState(prevState => ({
+            exportData: this.props.htmlGenerator(prevState.exportDescription),
```

Comment:
My understanding is that the same logic behind wanting to avoid `this.state` in `setState` input also applies to `this.props`. It's benign here because the `htmlGenerator` happens to be constant, but I think the better pattern would probably be to use the two-argument form of `setState`'s parameter, eg:

```ts
this.setState((prevState, prevProps) => ({
    exportData: prevProps.htmlGenerator(prevState.exportDescription),
```